kong/dao/schemas/plugins_configurations.lua

```diff
@@ -38,6 +38,13 @@ return {
       return false, DaoError("No consumer can be configured for that plugin", constants.DATABASE_ERROR_TYPES.SCHEMA)
     end
 
+    if value_schema.self_check then
+      local ok, err = value_schema.self_check(value_schema, plugin_t.value and plugin_t.value or {}, dao, is_update)
+      if not ok then
+        return false, DaoError(err, constants.DATABASE_ERROR_TYPES.SCHEMA)
+      end
+    end
+
```

Comment:
@thibaultCha to fix the problem I had to implement this check. Basically this is the problem:

* BaseDao invokes the `validate_entity`. The entity is the `plugin_configuration`, which has a `self_check`.
* The `self_check` is never being invoked on the underlying plugin's schema.
* This code invokes the underlying plugins's `self_check` if it exists.